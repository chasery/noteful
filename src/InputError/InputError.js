import React from "react";
import PropTypes from "prop-types";
import "./InputError.css";

export default function ValidationError(props) {
  if (props.message) {
    return <div className="Error">{props.message}</div>;
  }
  return <></>;
}

ValidationError.defaultProps = {
  message: "",
};

ValidationError.propTypes = {
  message: PropTypes.string.isRequired,
};
