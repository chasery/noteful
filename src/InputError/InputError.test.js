import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import InputError from "./InputError";

describe("InputError component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<InputError />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
