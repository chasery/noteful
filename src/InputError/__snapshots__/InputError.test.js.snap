// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError component renders without error 1`] = `<Fragment />`;
